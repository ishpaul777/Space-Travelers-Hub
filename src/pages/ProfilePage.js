import './Styles/ProfilePage.css';

const ProfilePage = () => (
  <section className="profile--info--container flex">
    <div className="missions--joined--info">
      <h2>My Missions</h2>
      <table>
        <tr>
          <td>Random 1</td>
        </tr>
        <tr>
          <td>Random 2</td>
        </tr>
        <tr>
          <td>Random 3</td>
        </tr>
        <tr>
          <td>Random 4</td>
        </tr>
      </table>
    </div>
    <div className="rockets--booked--info">
      <h2>My Rockets</h2>
      <table>
        <tr>
          <td>Random 1</td>
        </tr>
        <tr>
          <td>Random 2</td>
        </tr>
        <tr>
          <td>Random 3</td>
        </tr>
      </table>
    </div>
  </section>
);

export default ProfilePage;
