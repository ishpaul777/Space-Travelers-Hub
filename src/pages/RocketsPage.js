import RocketsContainer from '../Components/rockets/RocketContainer';

const RocketsPage = () => (
  <div>
    <RocketsContainer />
  </div>
);

export default RocketsPage;
