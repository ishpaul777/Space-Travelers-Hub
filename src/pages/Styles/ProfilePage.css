.profile--info--container {
  margin-top: 80px;
  align-items: flex-start;
  justify-content: center;
  gap: 10vw;
  flex-wrap: wrap;
}

.profile--info--container table {
  margin-top: 20px;
  border-collapse: collapse;
  width: 30vw;
}

.profile--info--container td {
  border: 1px solid #868686;
  text-align: left;
  padding: 13px;
}
