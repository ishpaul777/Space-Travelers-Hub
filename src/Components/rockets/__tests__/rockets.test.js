import {
  fireEvent, render, screen,
} from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { combineReducers, configureStore } from '@reduxjs/toolkit';
import ProfilePage from '../../../pages/ProfilePage';
import missionReducer from '../../../redux/Missions/missions';
import rocketReducer from '../../../redux/Rockets/rockets';
import RocketsPage from '../../../pages/RocketsPage';

const initialState = {
  missions: [],
  rockets: [{
    id: 'falcon1',
    name: 'Falcon 1',
    isReserved: false,
    description: 'The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.',
    image: 'image--rocket1.jpg',
  }],
};

const rootReducer = combineReducers({
  missions: missionReducer,
  rockets: rocketReducer,
});

const store = configureStore({
  reducer: rootReducer,
  preloadedState: initialState,
});

describe('Rockets tests', () => {
  test('should render', () => {
    const rockets = render(
      <Provider store={store}>
        <BrowserRouter>
          <RocketsPage />
        </BrowserRouter>
        ,
      </Provider>,
    );

    expect(rockets).toMatchSnapshot();
  });
  test('Displays the Rocket', () => {
    render(
      <Provider store={store}>
        <RocketsPage />
      </Provider>,
    );
    const rocketName = screen.getByText('Falcon 1');
    expect(rocketName).toBeInTheDocument();
  });
  test('Reserving and reflecting the reserved rocket in Profile Page', () => {
    render(
      <Provider store={store}>
        <RocketsPage />
        <ProfilePage />
      </Provider>,
    );
    expect(screen.getByText('No Missions Joined')).toBeInTheDocument();

    fireEvent.click(screen.getByText('Reserve Rocket'));
    expect(screen.getAllByText('Falcon 1').length).toEqual(2);
  });

  test('Aborting a Rocket Reservation and removing the canceled rocket in Profile Page', () => {
    render(
      <Provider store={store}>
        <RocketsPage />
        <ProfilePage />
      </Provider>,
    );

    fireEvent.click(screen.getByText('Cancel Reservation'));
    expect(screen.getAllByText('Falcon 1').length).toEqual(1);
  });
});
