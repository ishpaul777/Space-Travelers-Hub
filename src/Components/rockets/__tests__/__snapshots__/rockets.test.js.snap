// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets tests should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <section
          class="rocket--container flex"
        >
          <div
            class="rocket flex"
          >
            <img
              alt="rocket"
              class="rocket--image"
              src="image--rocket1.jpg"
            />
            <div
              class="rocket--info"
            >
              <h2
                class="rocket--name"
              >
                Falcon 1
              </h2>
              <p
                class="rocket--description"
              >
                The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
              </p>
              <div
                class="user--actions"
              >
                <button
                  class="reserve--button"
                  type="button"
                >
                  Reserve Rocket
                </button>
              </div>
            </div>
          </div>
        </section>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div>
      <section
        class="rocket--container flex"
      >
        <div
          class="rocket flex"
        >
          <img
            alt="rocket"
            class="rocket--image"
            src="image--rocket1.jpg"
          />
          <div
            class="rocket--info"
          >
            <h2
              class="rocket--name"
            >
              Falcon 1
            </h2>
            <p
              class="rocket--description"
            >
              The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
            </p>
            <div
              class="user--actions"
            >
              <button
                class="reserve--button"
                type="button"
              >
                Reserve Rocket
              </button>
            </div>
          </div>
        </div>
      </section>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
